pkg/control/sorting/pod_sorting.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2021 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sorting
+
+import (
+	"context"
+	"fmt"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	appsv1beta1 "github.com/openkruise/kruise/apis/apps/v1beta1"
+	clonesetcore "github.com/openkruise/kruise/pkg/controller/cloneset/core"
+	synccontrol "github.com/openkruise/kruise/pkg/controller/cloneset/sync"
+	sidecarsetcontroller "github.com/openkruise/kruise/pkg/controller/sidecarset"
+	statefulsetcontroller "github.com/openkruise/kruise/pkg/controller/statefulset"
+	"github.com/openkruise/kruise/pkg/util/updatesort"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SortPods sorts the given Pods according the owner workload logic.
+func SortPods(reader client.Reader, ns string, owner metav1.OwnerReference, pods []*v1.Pod) ([]*v1.Pod, error) {
```

Comment:
pls comments the usage of the sorting, only for image predownload function?

Reply:
done